src/script/lua_api/l_object.cpp

```diff
@@ -684,6 +684,85 @@ int ObjectRef::l_get_properties(lua_State *L)
 	return 1;
 }
 
+// set_observers(self, observers)
+int ObjectRef::l_set_observers(lua_State *L)
+{
+	GET_ENV_PTR;
+	ObjectRef *ref = checkObject<ObjectRef>(L, 1);
+	ServerActiveObject *sao = getobject(ref);
+	if (sao == nullptr)
+		throw LuaError("Invalid ObjectRef");
```

Comment:
by convention we just `return 0` here

Reply:
I know, but I think it's not a good convention. If the engine can't fulfill the request to set something because it was provided with an invalid ObjRef, it shouldn't silently return nothing, sweeping bugs under the rug; it should loudly complain.

However, this arguably comes at the cost of consistency. So the question boils down to:
Does being consistent outweigh being "strict" here? I'd say no.

Reply:
I say keep it at `return 0` and changes can be looked at later.
Perhaps making all setters throw and getters silently return `nil`.

Reply:
There is another reason to not return `nil` here: `nil` is a valid return value. So I would prefer to keep the error rather than overloading the `nil` return.